src/libraries/System.Private.CoreLib/src/System/DateTimeOffset.Android.cs

```diff
@@ -0,0 +1,76 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Threading;
+
+namespace System
+{
+    public readonly partial struct DateTimeOffset
+    {
+        // 0 == in process of being loaded, 1 == loaded
+        private static int s_androidTZDataLoaded = -1;
+
+        // Now on Android does the following
+        // 1) quickly returning a fast path result when first called if the right AppContext data element is set
+        // 2) starting a background thread to load TimeZoneInfo local cache
+        //
+        // On Android, loading AndroidTZData is expensive for startup performance.
+        // The fast result relies on `System.TimeZoneInfo.LocalDateTimeOffset` being set
+        // in the App Context's properties as the appropriate local date time offset from UTC.
+        // monovm_initialize(_preparsed) can be leveraged to do so.
+        // However, to handle timezone changes during the app lifetime, AndroidTZData needs to be loaded.
+        // So, on first call, we return the fast path and start a background thread to load
+        // the TimeZoneInfo Local cache implementation which loads AndroidTZData.
+        public static DateTimeOffset Now
+        {
+            get
+            {
+                DateTime utcDateTime = DateTime.UtcNow;
+                object? localDateTimeOffset;
+
+                if (s_androidTZDataLoaded == 1) // The background thread finished, the cache is loaded.
```

Comment:
Technically, this should probably be:
```C#
if (Volatile.Read(ref s_androidTZDataLoaded) == 1)
```
at which point you could just make the static field be `volatile` and get rid of the `Volatile.Read/Write` at each call site.  I don't think it'll matter in practice.  But if we're going for as simple as possible, just making the field be volatile would do so.